src/NumericalAlgorithms/Spectral/SwshDerivatives.hpp

```diff
@@ -305,5 +305,168 @@ using SwshDerivatives = detail::SwshDerivativesImpl<
+ * \brief Evaluate all of the spin-weighted derivatives in `DerivKindList` on
+ * input `SpinWeighted<ComplexDataVector, Spin>` collocation data, returning by
+ * pointer.
+ *
+ * \details This function provides two interfaces, one in which the caller
+ * provides the intermediate coefficient buffers needed during the computation
+ * of the derivatives, and one in which those buffers are temporarily allocated
+ * during the derivative function calls.
+ *
+ * For the interface in which the caller does not provide buffers, the arguments
+ * must take the following structure (enforced by internal function calls):
+ *
+ * - `size_t l_max` : angular resolution for the spherical representation
+ * - `size_t number_of_radial_points` : radial resolution (number of consecutive
+ * blocks to evaluate derivatives, for each input vector )
+ * - for each `DerivKind` in `DerivKindList`, a
+ * `gsl::not_null<SpinWeighted<ComplexDataVector, Spin +
+ * Tags::derivative_spin_weight<DerivKind>>>` : the output of the derivative
+ * evaluation
+ * - for each `DerivKind` in `DerivKindList`, a `const
+ * SpinWeighted<ComplexDataVector, Spin>` (where the `Spin` for these arguments
+ * matches the corresponding vector from the previous set) : the input to the
+ * derivative evaluation.
+ *
+ * For the interface in which the caller does provide buffers, the arguments
+ * must take the following structure (enforced by internal function calls):
+ *
+ * - `size_t l_max` : angular resolution for the spherical representation
+ * - `size_t number_of_radial_points` : radial resolution (number of consecutive
+ * blocks to evaluate derivatives, for each input vector )
+ * - for each `DerivKind` in `DerivKindList`, a
+ * `gsl::not_null<SpinWeighted<ComplexModalVector, Spin +
+ * Tags::derivative_spin_weight<DerivKind>>>` : the buffer for the spin-weighted
+ * spherical harmonic modes of the derivative quantities.
+ * - for each `DerivKind` in `DerivKindList`, a `const
+ * SpinWeighted<ComplexModalVector, Spin>` (where the `Spin` for these arguments
+ * matches the corresponding vector from the previous set) : the buffer for the
+ * spin-weighted spherical harmonic modes of the input quantities.
+ * - for each `DerivKind` in `DerivKindList`, a
+ * `gsl::not_null<SpinWeighted<ComplexDataVector, Spin +
+ * Tags::derivative_spin_weight<DerivKind>>>` : the output of the derivative
+ * evaluation
+ * - for each `DerivKind` in `DerivKindList`, a `const
+ * SpinWeighted<ComplexDataVector, Spin>` (where the `Spin` for these arguments
+ * matches the corresponding vector from the previous set) : the input to the
+ * derivative evaluation.
+ *
+ * The function `swsh_buffer` assists in generating the modal buffers of
+ * appropriate size.
+ */
+template <
+    typename DerivativeKindList,
+    ComplexRepresentation Representation = ComplexRepresentation::Interleaved,
+    typename... ArgumentTypes>
+void swsh_derivatives(const size_t l_max, const size_t number_of_radial_points,
```

Comment:
why does this function need the `swsh_` prefix if it's already in the `swsh` namespace?

Reply:
My reasoning was to ensure that it is clear that this is taking angular derivatives using the harmonic representations, rather than (for instance) volume derivatives. If this name doesn't make that any clearer, I can omit the `swsh` on all of the public utilities in this file.

Reply:
That wasn't clear to me from the name, though it was clear to me from the namespace. However, I don't think it's bad to have additional clarity. What do you think of naming the function `angular_derivatives`?